File: ELEC881_Copilot/src/Function Name Masked/Java script (JS)/q7-super-ugly-number.js
var test = function(n) {
    var primes = [2, 7, 13, 19];
    var ugly = [1];
    var index = new Array(primes.length).fill(0);
    var val = new Array(primes.length).fill(1);
    for (var i = 1; i < n; i++) {
        for (var j = 0; j < primes.length; j++) {
            val[j] = ugly[index[j]] * primes[j];
        }
        var min = Math.min.apply(null, val);
        for (var j = 0; j < primes.length; j++) {
            if (min === val[j]) {
                index[j]++;
            }
        }
        ugly.push(min);
    }
    return ugly[n - 1];

};

File: ELEC881_Copilot/src/Function Name Masked/Python/q7-super-ugly-number.py
class test(object):
    def test1(self, n, primes):
        ugly = [1]
        idx = [0] * len(primes)
        while len(ugly) < n:
            next = min([ugly[idx[i]] * primes[i] for i in range(len(primes))])
            for i in range(len(primes)):
                if ugly[idx[i]] * primes[i] == next:
                    idx[i] += 1
            ugly.append(next)
        return ugly[-1]